src/nxt_http_return.c

```diff
@@ -51,20 +50,18 @@ nxt_http_return_init(nxt_mp_t *mp, nxt_http_action_t *action,
 
     conf->status = nxt_conf_get_number(acf->ret);
 
-    if (acf->location.length == 0) {
-        conf->loc_is_const = 1;
+    if (acf->location == NULL) {
         return NXT_OK;
     }
 
-    var = nxt_var_compile(&acf->location, mp, 0);
-    if (nxt_slow_path(var == NULL)) {
+    nxt_conf_get_string(acf->location, &str);
+
+    conf->location = nxt_var_compile(&str, mp, 0);
+    if (nxt_slow_path(conf->location == NULL)) {
         return NXT_ERROR;
     }
 
-    conf->location = var;
-    conf->loc_is_const = nxt_var_is_const(var);
-
-    if (conf->loc_is_const) {
+    if (nxt_var_is_const(conf->location)) {
```

Comment:
Tiny optimization:
```
    if (nxt_var_is_const(conf->location)) {
        nxt_var_raw(conf->location, &str);
        if (str.length > 0) {
            return nxt_http_return_encode(mp, &conf->encoded, &str);
       }
    }

return NXT_OK;
```

Reply:
That would optimize for the following case:

```json
"action": {
    "return": 301,
    "location": "$host"
}
```

at the cost of adding a conditional (an operation or so) in the common case.

Since that case is quite useless, I prefer keeping it simple, and optimize for the common case.  Anyway, `nxt_http_return_encode()` will return quite fast when the string is empty.

Reply:
Yeah, we don't need to do it.

```
static nxt_int_t
nxt_http_return_encode(nxt_mp_t *mp, nxt_str_t *encoded,
    const nxt_str_t *location)
{
    nxt_uint_t  encode;

    if (nxt_is_complex_uri_encoded(location->start, location->length)) {
        // it contains the case of empty string.
        *encoded = *location;

        return NXT_OK;
    }
```